crates/pathfinder/src/bin/pathfinder/main.rs

```diff
@@ -215,12 +215,25 @@ Hint: This is usually caused by exceeding the file descriptor limit of your syst
         verify_tree_hashes: config.verify_tree_hashes,
     };
 
-    let sync_handle = tokio::spawn(state::sync(sync_context, state::l1::sync, state::l2::sync));
+    let is_sync_enabled = config.is_sync_enabled;
+    let sync_handle = tokio::spawn(async move {
+        if is_sync_enabled {
+            state::sync(sync_context, state::l1::sync, state::l2::sync).await
+        } else {
+            Ok(())
+        }
+    });
```

Comment:
I think either make it an optional join-handle, or have the else be forever pending? Otherwise you have to do weird logic elsewhere which isn't obvious. e.g. the `select!` lower down.